src/garage/envs/gym_env.py

```diff
@@ -242,6 +244,9 @@ def step(self, action):
         if step_type in (StepType.TERMINAL, StepType.TIMEOUT):
             self._step_cnt = None
 
+        if self.spec.observation_space.flat_dim != np.prod(observation.shape):
```

Comment:
Just curious why comparing the flat_dim instead of the original shape.

Reply:
It's pretty common for environments to give `observation_space` as a finite box, then exceed that box. Having said that, this change should probably instead do a check more like the one in `EpisodeBatch`: first check if `self.spec.observation_space.contains(observation)`, and then only check the flat dimensions if that's false (this matters for discrete observations).